### PR TITLE
LPS-132411 As a content manager, I want to filter by Type and/or Subtype of content in the Content Dashboard

### PR DESCRIPTION
**Pull request by 4lejandrito**
[Jira issue](https://issues.liferay.com/browse/LPS-132411)

### CODE REVIEW

**Comment by 4lejandrito** [comment]
See https://github.com/liferay-tango/liferay-portal/pull/1486.

File: modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/SelectTypeAndSubtype.js [hunk 1]
```diff
@@ -18,33 +18,39 @@ import {Treeview} from 'frontend-js-components-web';
...
-			id: node.classPK || `_${index}`,
+			expanded: !!(!index && hasChildren) || false,
+			id: _getNodeId({index, node}),
 			name: node.label,
 		};
 	});
 };
 
-function visit(nodes, callback) {
+const visit = (nodes, callback) => {
```
Comment (sandrodw3) [inline modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/SelectTypeAndSubtype.js:43]: And this too 😛

**Review by sandrodw3 — COMMENTED** [review]
I love this PR! 😍

Great frontend work @claraizquierdo @beltranrengifo! 🌈
I only added some tiny comments, most of them related to format and my obsesion with a clean commit history 😂, sorry if I insist a lot on this.

You two make a great front team 🥰

File: modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/SelectTypeAndSubtype.js [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+}
+
+const handleSelectionChange = (selectedNodes) => {
+    // TODO
+};
+
+function SelectTypeAndSubtype() {
+    /* Mocked values */
+    const namespace = "mynamespace";
+    const nodes = [
```
[the remark this answers is not in the document]
Comment (beltranrengifo) [reply modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/SelectTypeAndSubtype.js]: I believe Clara had to commit this so I could continue with the task while the backend was in progress

File: modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/view_content_dashboard_item_types.jsp [hunk 3]
```diff
@@ -22,7 +22,11 @@ ContentDashboardItemSubtypeItemSelectorViewManagementToolbarDisplayContext conte
 ContentDashboardItemSubtypeItemSelectorViewDisplayContext contentDashboardItemSubtypeItemSelectorViewDisplayContext = (ContentDashboardItemSubtypeItemSelectorViewDisplayContext)request.getAttribute(ContentDashboardItemSubtypeItemSelectorViewDisplayContext.class.getName());
 %>
 
-<clay:management-toolbar
+<react:component
+	module="js/SelectTypeAndSubtype"
+/>
+
+<!-- <clay:management-toolbar
```
[the remark this answers is not in the document]
Comment (beltranrengifo) [reply modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/view_content_dashboard_item_types.jsp]: Agree!

File: modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/SelectTypeAndSubtype.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (beltranrengifo) [reply modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/SelectTypeAndSubtype.js]: I prefer to leave the mock in this case, this way the first two commits of Clara will still work, until the props arrive later

File: modules/apps/content-dashboard/content-dashboard-web/src/main/resources/content/Language.properties [hunk 4]
```diff
@@ -17,6 +17,7 @@ in-use=In Use
 javax.portlet.title.com_liferay_content_dashboard_web_portlet_ContentDashboardAdminPortlet=Content Dashboard
 languages-translated-into=Languages Translated Into
 metrics=Metrics
+no-types-were-found=No types were found.
```
[the remark this answers is not in the document]
Comment (beltranrengifo) [reply modules/apps/content-dashboard/content-dashboard-web/src/main/resources/content/Language.properties:21]: I agree but I really don't know when this string was added 😛

File: modules/apps/content-dashboard/content-dashboard-web/test/js/components/AuditBarChart.js [hunk 5]
```diff
@@ -226,7 +226,11 @@ describe('AuditBarChart', () => {
 
 	it('renders audit bar chart from one vocabulary', () => {
 		const {container, getByText} = render(
-			<AuditBarChart rtl={false} vocabularies={mockOneVocabulary} />
```
[the remark this answers is not in the document]
Comment (beltranrengifo) [reply modules/apps/content-dashboard/content-dashboard-web/test/js/components/AuditBarChart.js:229]: Love it!

**Comment by beltranrengifo** [comment]
ci:forward

**Comment by beltranrengifo** [comment]
ci:forward

**Comment by sandrodw3** [comment]
Failing tests look related @beltranrengifo 😛
Maybe @CarlosBrichete can help!

**Comment by beltranrengifo** [comment]
> Failing tests look related @beltranrengifo 😛
> Maybe @CarlosBrichete can help!

Related between them?
Uhmm gonna give it a try again... 🤣

**Comment by beltranrengifo** [comment]
ci:forward

**Comment by sandrodw3** [comment]
> > Failing tests look related @beltranrengifo 😛
> > Maybe @CarlosBrichete can help!
> 
> Related between them?
> Uhmm gonna give it a try again... 🤣

I mean to your changes, they are related to Content Dashboard, so probably they need to be fixed!

**Comment by 4lejandrito** [comment]
ci:forward

**Comment by 4lejandrito** [comment]
ci:forward

**Comment by 4lejandrito** [comment]
ci:forward

**Pull request closed without merging**